### ISSUE TITLE
POST-Parameters omitted in AJAX request when using execute with id of parent component

### ISSUE THREAD

**Issue by flauto150** [omnifaces/omnifaces#810]
Hi, 

I use:
* jakarta-faces: 2.3.21
* omnifaces: 3.14.5
* weld-servlet-shaded: 3.1.9
* Tomcat 9.0
* Bootstrap 4.0 with jquery

I try to evaluate the effort to migrate a big JSF 2.2 (with omnifaces 1.14.1) application to JSF 2.3 and have the following problem:

We often have ajax requests, where we use the id of a <h:panelGroup>-Element as execute-Parameter to execute the complete tree below this element.
Here a simplified example of the structure:
```

<html
	xmlns="http://www.w3.org/1999/xhtml"
	xmlns:h="http://xmlns.jcp.org/jsf/html"
	xmlns:ui="http://xmlns.jcp.org/jsf/facelets"
	xmlns:pa="http://xmlns.jcp.org/jsf/passthrough"
	xmlns:o="http://omnifaces.org/ui">
	(...)
<o:form id="filterForm">
	(...)
  <(own composite component with id="filterPanel")>
	<h:panelGroup id="filterContainer">
		<h:panelGroup id="filterRow">
			<h:selectOneMenu id="filterField" .../>
			<h:selectOneMenu id="filterCondition" .../>
			<h:panelGroup id="filterValue" ...>
				<h:commandButton id="addFilter">
					<f:ajax
						listener="#{someListener}"
						execute="filterRow"
						render="otherids">
					</f:ajax>
		</h:panelGroup>
	</h:panelGroup>
  </(own composite component)
</o:form>
</html>
```
		
In this example the child components below filterRow (filterField, filterCondition,...) are not sent as POST parameters in the AJAX Request with JSF 2.3/omnifaces 3.14.5. With JSF 2.2/omnifaces 1.14.1  they were sent. When I use `@form` instead of `fiterRow `as execute parameter, everything is working fine. When I don't use omnifaces (`<h:form>` instead of `<o:form>`), everything is fine too.

I've debugged this in the Javascript-Code and found that the problem will be gone, if I change in https://github.com/omnifaces/omnifaces/blob/3.x/src/main/resources/META-INF/resources/omnifaces/form.unminified.js in line 78 
`var name = key.replace("%3A", "\\:"); `
to
`var name = key.replaceAll("%3A", "\\:");`

Then all separator characters in the absolute ID of the execute element (`filterForm:filterPanel:filterRow`) are unescaped here and the child elements are found.
With `replace `only the first separator character is replaced (`filterForm\:filterPanel%3AloadedFilter`) and the child elements are not found.

Is there anything wrong in my understanding? Or is this perhaps something, that could be fixed in omnifaces?

Kind regards
Corinna

**Comment by pizzi80** [comment]
Yes, your analysis is correct! 

it's probably an error due to the fact that in Java `String.replace` 
replaces every string with the passed replacement 
while in JS only the first occurrence is replaced...

**Comment by flauto150** [comment]
Thanks for the reply.
Is it also planned/possible to fix this in omnifaces3? Currently I'm using a locally patched version...

**Comment by BalusC** [comment]
The fix will be in the upcoming 3.14.6 / 4.5.
